apollo-federation/src/connectors/json_selection/apply_to.rs

```diff
@@ -561,35 +561,30 @@ impl ApplyToInternal for WithRange<PathList> {
                             tail.apply_to_path(shallow_mapped_array, vars, &input_path_with_key)
                         })
                 } else {
+                    let not_found = (
+                        None,
+                        vec![ApplyToError::new(
+                            format!(
+                                "Property {} not found in {}",
+                                key.dotted(),
+                                json_type_name(data),
+                            ),
+                            input_path_with_key.to_vec(),
+                            key.range(),
+                        )],
+                    );
```

Comment:
Making this error is not free, and we don't want to allocate the two vecs and two strings for every path step even if there wasn't an error.

Sooo, if you want to deduplicate the code, you could make this a closure?

```suggestion
                    let not_found = || (
                        None,
                        vec![ApplyToError::new(
                            format!(
                                "Property {} not found in {}",
                                key.dotted(),
                                json_type_name(data),
                            ),
                            input_path_with_key.to_vec(),
                            key.range(),
                        )],
                    );
```

And then just add the calling parens after each invocation down below, like `not_found()`